text/0010-exception-types.md

```diff
@@ -0,0 +1,120 @@
+
+Sentry shows exceptions which aren't caught by the developer (unhandled) but also do not cause the software to exit in the same way as exceptions which are manually caught by the developer. This seems rather unintuitive and makes exceptions seem less severe than they are.
+
+This issue impacts for example the [Dart/Flutter SDK](https://github.com/getsentry/sentry-dart/issues/456), the Unity SDK, the React Native SDK and possibly more.
+
+Another issue is, that excpetions which don't cause the software to exit but are unhandled, are not considered in the [`session health` metric](https://develop.sentry.dev/sdk/sessions/).
+Currently, the session would be marked as `errored` instead of `crashed`.
+
+The attribute `thread.errored` was added in the past for similar reasons, but it got [reverted on Relay](https://github.com/getsentry/relay/pull/306) and [Android](https://github.com/getsentry/sentry-android/pull/187).
+
+# Option 1 (recommended)
+
+Based on the problem stated above, I propose to introduce the types of `handled`, `unhandled`, `process termination` (this is the same as the current `handled`, but rephrased to avoid confusion). I'm open for better phrasing of those types, but I'll stick to those names for the rest of the RFC. The meaning of those types is as follows:
+
+- `handled`: The exception was recorded by a developer via `Sentry.capture*` method. May or may not be visually indicated by the Sentry user interface.
+- `unhandled`: Indicates whether the exception was recorded automatically by Sentry through the use of a global exception handler or similar. This exception however didn't cause the software to exit, and the software will continue to be executed. This should be visualized in the Sentry user interface, they have a higher severity than the `handled` ones.
+- `process termination`: The exception was recorded automatically by Sentry through the use of a exception handler or similar. The exception caused the software to terminate the execution. This should be visualized in the Sentry user interface. This is currently done by the `handled: false` flag in the [exception mechanism](https://develop.sentry.dev/sdk/event-payloads/exception/#exception-mechanism).
+
+A user of Sentry should be able to 
+
+- filter events on the issues page or discover for the newly introduces exception types (3 categories).
+- highlight (similar to the unhandled label) events of the type unhandled and process termination.
+- get alerted for events of the type unhandled and process termination separately.
+
+Currently, there's an unhandled label on the issue's page but it's only highlighted for process termination errors (if `handled: false`).
+
+In order to propagate those exception types, the exception mechanism needs to be adapted:
```

Comment:
I like this option more, because it is backwards compatible. 
But I would not add another boolean, the new "process_terminated" variable needs to be added because "handled" is a boolean and thus not very future proof. 
So I would add a number/enum kind of type called something more general like `error_state` or such that can then have values like `process_terminated`, `timeout`, `deadlock`, `caught-fire`, ...

Reply:
> One general question: How can we send data to sentry, when the process terminates? Is this even possible? (can we store the error fast enough to disk to send on next start?)

yes